src/client/components/configuration/tree_store.tsx

```diff
@@ -0,0 +1,39 @@
+import { action, observable } from 'mobx'
+import { Node } from './tree/tree'
+
+export default class TreeStore {
+  @observable
+  public data: Node
+
+  @observable
+  public selectedNode: Node
+
+  constructor(data: Node) {
+    this.data = data
+    this.onNodeClick = this.onNodeClick.bind(this)
+  }
+
+  @action
+  public onNodeClick(node: Node): void {
```

Comment:
One of my architectural goals is to keep models as dumb as possible, such that they are _just_ data and no behaviour.

That means actions like these need to live somewhere else, preferably somewhere that we can test.

This is why I created a `LocalisationPresenter` for the localisation component, and any `@action` lives as a method on the presenter, such that the `LocalisationModel` itself does not have any actions on it. I'll soon be renaming `LocalisationPresenter` to `LocalisationController` as well.

Could we follow a similar pattern here? 😄  You'll probably want a separate `ConfigurationController` class where these actions can live 🙂